hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -412,19 +412,21 @@ protected boolean isFileSliceAfterPendingCompaction(FileSlice fileSlice) {
    * With async compaction, it is possible to see partial/complete base-files due to inflight-compactions, Ignore those
    * base-files.
    *
-   * @param fileSlice File Slice
+   * @param fileSliceStream Stream of FileSlice
+   * @param includeEmptyFileSlice include empty file-slices
    */
-  protected FileSlice filterBaseFileAfterPendingCompaction(FileSlice fileSlice) {
-    if (isFileSliceAfterPendingCompaction(fileSlice)) {
-      LOG.debug("File Slice (" + fileSlice + ") is in pending compaction");
-      // Base file is filtered out of the file-slice as the corresponding compaction
-      // instant not completed yet.
-      FileSlice transformed =
-          new FileSlice(fileSlice.getPartitionPath(), fileSlice.getBaseInstantTime(), fileSlice.getFileId());
-      fileSlice.getLogFiles().forEach(transformed::addLogFile);
-      return transformed;
-    }
-    return fileSlice;
+  protected Stream<FileSlice> filterBaseFileAfterPendingCompaction(Stream<FileSlice> fileSliceStream, boolean includeEmptyFileSlice) {
+    return fileSliceStream.map(fileSlice -> {
+      if (isFileSliceAfterPendingCompaction(fileSlice)) {
+        LOG.debug("File Slice (" + fileSlice + ") is in pending compaction");
+        // Base file is filtered out of the file-slice as the corresponding compaction
+        // instant not completed yet.
+        FileSlice transformed = new FileSlice(fileSlice.getPartitionPath(), fileSlice.getBaseInstantTime(), fileSlice.getFileId());
+        fileSlice.getLogFiles().forEach(transformed::addLogFile);
+        return transformed;
+      }
+      return fileSlice;
+    }).filter(slice -> includeEmptyFileSlice || !slice.isEmpty());
```

Comment:
We don't need separate filtering instead:

 - Replace above `map` into `flatMap`
 - If `includeEmptyFileSlice` is false return `Stream.of()`